crates/primitives/receipt/src/lib.rs

```diff
@@ -20,6 +20,36 @@ pub enum TransactionReceipt {
     DeployAccount(DeployAccountTransactionReceipt),
 }
 
+impl From<InvokeTransactionReceipt> for TransactionReceipt {
+    fn from(receipt: InvokeTransactionReceipt) -> Self {
+        TransactionReceipt::Invoke(receipt)
+    }
+}
+
+impl From<L1HandlerTransactionReceipt> for TransactionReceipt {
+    fn from(receipt: L1HandlerTransactionReceipt) -> Self {
+        TransactionReceipt::L1Handler(receipt)
+    }
+}
+
+impl From<DeclareTransactionReceipt> for TransactionReceipt {
+    fn from(receipt: DeclareTransactionReceipt) -> Self {
+        TransactionReceipt::Declare(receipt)
+    }
+}
+
+impl From<DeployTransactionReceipt> for TransactionReceipt {
+    fn from(receipt: DeployTransactionReceipt) -> Self {
+        TransactionReceipt::Deploy(receipt)
+    }
+}
+
+impl From<DeployAccountTransactionReceipt> for TransactionReceipt {
+    fn from(receipt: DeployAccountTransactionReceipt) -> Self {
+        TransactionReceipt::DeployAccount(receipt)
+    }
+}
```

Comment:
wdyt about implementing `AsRef` too?

Reply:
do you think a `TryAsRef` on the enum or `AsRef` on the methods below?

Reply:
hm `TryAsRef` should work there

Reply:
ok I will add that in #227 as it concerns primitives